db/migrate/20240617163508_create_journeys.rb

```diff
@@ -0,0 +1,10 @@
+class CreateJourneys < ActiveRecord::Migration[7.1]
+  def change
+    create_table :journeys do |t|
+      t.integer :annual_miles, null: false
```

Comment:
⛏️ Why integers? What if my goal is a marathon and I want 36.2 miles?

Reply:
The journeys are uploaded by an admin and are based on a TN landmark. Example: I-40 across the state of TN = 455 miles. The actual distance is 455.28 but we've just rounded them off to make things easier. So the user doesn't have the ability to create their own journey or distance. The only reason why I am using monthly and weekly rates is because the monthly and weekly do not always add up to be the annual amount (as dictated to me by the client). It may be off by a little bit. The annual distance is the only one we are using to benchmark their progress. The monthly and weekly are just to display to the user when they are choosing a journey so they can get an idea of how much they are committing to per month/week. Hope that makes sense

Reply:
It does, but you may have an easier time if you at least make them a :float. Ruby can convert between a Float and an Integer on the fly, but we'd prefer to not do that conversion at runtime. The User won't be walking in even increments of 1 mile, so one will always need to be converted.
You also get weird behavior with some operations if you don't use a float:
```ruby
$ 2/216
 => 0 # should be .00989...
 
 $ 17/4
  => 4 # should be 4.25

def find_avg(list)
  list.sum / list.size
end

find_avg([5,6,8,4])
> 5 # should be 5.75
```

Reply:
Okay that totally makes sense. Originally I was storing mileage as hundredths so that I wouldn't run into any issues like that. It was a bit confusing though because I was constantly having to multiply or divide the number by 100. Do you think it's a better idea to store all the mileage as a float? Also looking back at the Route model, we might want to change step_count to mileage since the phone can give us a more accurate mileage reading than calculating based on steps.

Reply:
Storing the mileage as hundredths is not a bad idea, you get to avoid all the weirdness that comes with imprecise floats. This is especially good if you are dealing with money, you can store a dollar as 1000. You do have to remember to convert but you only need to do that when the data leaves your system as serialized JSON or something like that.

If the phone gives you mileage, we may as well use it since it'll be more accurate than using steps. 👍

Reply:
Yep! Personally I am a big fan of --fixup commit and --autosquash
https://thoughtbot.com/blog/autosquashing-git-commits

Reply:
Okay awesome!! Checking this out now